frontera/contrib/backends/cassandra/models.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+import uuid
+from cassandra.cqlengine.models import Model
+from cassandra.cqlengine.usertype import UserType
+from cassandra.cqlengine.columns import Map, Text, Float, Integer, DateTime, UserDefinedType, Counter, Boolean, \
+    SmallInt, BigInt
+
+
+class Meta(UserType):
+    domain = Map(Text(), Text(), required=False)
+    fingerprint = Text()
+    origin_is_frontier = Boolean()
+    scrapy_callback = Text()
+    scrapy_errback = Text()
+    scrapy_meta = Map(Text(), Text(), required=False)
+    score = Float(required=False)
+    jid = Integer(required=False)
+
+
+class MetadataModel(Model):
+    __table_name__ = 'metadata'
+
+    crawl = Text(primary_key=True)
+    fingerprint = Text(primary_key=True)
+    url = Text(index=True)
+    depth = Integer()
+    created_at = DateTime()
+    fetched_at = DateTime(required=False)
+    status_code = Integer(required=False)
+    score = Float(required=False)
+    error = Text(required=False)
+    meta = UserDefinedType(Meta)
+    headers = Map(Text(), Text(), required=False)
+    cookies = Map(Text(), Text(), required=False)
+    method = Text(required=False)
+
+    @classmethod
+    def query(cls, session):
+        return session.query(cls)
+
+    def __repr__(self):
+        return '<Metadata:%s (%s)>' % (self.url, self.fingerprint)
+
+
+class StateModel(Model):
+    __table_name__ = 'states'
+
+    crawl = Text(primary_key=True)
+    fingerprint = Text(primary_key=True)
+    state = SmallInt(index=True)
+
+    @classmethod
+    def query(cls, session):
+        return session.query(cls)
+
+    def __repr__(self):
+        return '<State:%s=%d>' % (self.fingerprint, self.state)
+
+
+class QueueModel(Model):
+    __table_name__ = 'queue'
+
+    crawl = Text(primary_key=True)
```

Comment:
Hi Alex,

it designed for:
1. Get next results
Here we need to filter after crawlid and partition_id.
Thats why they are at the first fields of the table/model.

2. Sorting get_next_results (for disributed mode: score, for non disributed mode: created_at)
Like i have written if you want to order by a column you must have the colums which are in order bevore the colum you like to sort in where or order by part of the query.
This: "SELECT * FROM queue WHERE crawl = 'default' AND partition_id=1 ORDER BY created_at" would not work bevause missing score.
If you put fingerprint bevor created_ar you must also define it in the query.
Thats because cassandra has a special way it stores data...

Also fingerprint must be in the clustering colum / primary key because of deletion with the key.

It's also possible to use indexes for some of the mentioned things - but they are slower and not recommendet for production usecases...

Hope i've  understand your question right.